cpp-client-reference/include/volition.h

```diff
@@ -0,0 +1,223 @@
+#pragma once
+#ifndef VOLITION_H
+#define VOLITION_H
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+#if __cplusplus >= 201103L || _MSVC_LANG >= 201103L
+#define VL_NULL_HANDLE nullptr
+#define VLAPI_NOEXCEPT noexcept
+#else
+#define VL_NULL_HANDLE NULL
+#define VLAPI_NOEXCEPT throw()
+#endif
+#else
+#define VL_NULL_HANDLE NULL
+#define VLAPI_NOEXCEPT
+#endif
+
+#ifdef _WIN32
+#define VLAPI_PUBLIC __declspec(dllexport)
+#elif __GNUC__ >= 4
+#define VLAPI_PUBLIC __attribute__((visibility("default")))
+#else
+#define VLAPI_PUBLIC
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Handle types.
+typedef struct vl_evaluable_s* vl_evaluable;
+typedef struct vl_event_stream_s* vl_event_stream;
+typedef struct vl_input_s* vl_input;
+typedef struct vl_optimizer_s* vl_optimizer;
```

Comment:
whats the purpose of this?

Reply:
These are type aliases. The idea is that the user can use types like `vl_evaluable` without having to know they are actually pointers (or handles).